### PR TITLE
New command to run WP-CLI commands on any site.

### PR DESCRIPTION
**Pull request by philipjohn**
Provides `vip sandbox run <site> <command> [--subcommand]` to enable the triggering of WP-CLI commands against a particular site from a sandbox, without the need to sandbox the site and enter a sandbox container.

It's a work-in-progress - I don't currently know what to do with the `runOnExistingContainer()` to check success/failure or anything else. We also need to get the output.

Attempt at #77

### CODE REVIEW

**Comment by joshbetz** [comment]
> vip sandbox run <site> <command> [--subcommand]

Rather than a subcommand option, I would do something like `vip sandbox run <site> <command...>` and join the command array and pass the whole thing as the command. So you could run `vip sandbox run v2.joshbetz.com wp site list` and `wp site list` would get passed as the command.

https://github.com/tj/commander.js/#variadic-arguments

**Comment by joshbetz** [comment]
> I don't currently know what to do with the runOnExistingContainer() to check success/failure or anything else. We also need to get the output.

What kind of failures are you trying to detect? For platform-level failures (site doesn't exist, couldn't create a sandbox container) we output an error inside `runOnExistingContainer()` and bail. If all that works, we fork to `docker exec` and stdout/stderr are sent to the terminal as if you were running the command directly.

**Comment by philipjohn** [comment]
> Rather than a subcommand option, I would do something like `vip sandbox run <site> <command...>` and join the command array and pass the whole thing as the command.

Okay, can do. I was copying the behaviour from the similar `execute-against-{...}` commands on wpcom, which take a `--subcommand` argument, but we can combine into one.

> What kind of failures are you trying to detect?

I'm mainly thinking of failures of the WP CLI command that we're running. Basically, I'm just thinking about catching stuff so there is some feedback to the person running the command.

**Comment by philipjohn** [comment]
So, `runOnExistingContainer()` is a bit of a black box to me - I'm not sure what that does and how my command can get the output from any commands it runs.

At the moment, when I run

> vip sandbox run 68 wp option get home

I see this:

> env: wp --allow-root option get home: No such file or directory

File: src/bin/vip-sandbox.js [hunk 1]
```diff
@@ -227,7 +227,10 @@ program
 				return console.error( 'Please provide a valid command to run.' );
 			}
 
-			// Re-build the WP CLI command
+			// Ensure we run as root. This assumes the user has included the 'wp'.
+			command[0] = 'wp --allow-root';
```
Comment (joshbetz) [inline src/bin/vip-sandbox.js]: I don't think we should blindly pass `--allow-root`. There's another PR to start containers as a different user, which is probably a better way to deal with this https://github.com/Automattic/vip-cli/pull/132

**Comment by joshbetz** [comment]
Fixed in #151

**Pull request closed without merging**